crates/mipsy_interactive/src/interactive/commands/mod.rs

```diff
@@ -1,115 +1,217 @@
-mod breakpoint;
 #[allow(clippy::module_inception)]
-mod commands;
-mod context;
-mod disassemble;
-mod dot;
-mod examine;
-mod exit;
-mod help;
-mod label;
-mod labels;
-mod load;
-mod print;
-mod reset;
-mod run;
-mod step;
 pub(crate) mod util;
-mod watchpoint;
-
-pub(crate) use breakpoint::breakpoint_command;
-pub(crate) use context::context_command;
-pub(crate) use disassemble::disassemble_command;
-pub(crate) use dot::dot_command;
-pub(crate) use examine::examine_command;
-pub(crate) use exit::exit_command;
-pub(crate) use help::help_command;
-pub(crate) use label::label_command;
-pub(crate) use labels::labels_command;
-pub(crate) use load::load_command;
-pub(crate) use print::print_command;
-pub(crate) use reset::reset_command;
-pub(crate) use run::run_command;
-pub(crate) use step::step_command;
-pub(crate) use watchpoint::watchpoint_command;
+
+use std::fmt::Display;
+
+pub(super) mod breakpoint;
+pub(super) mod commands;
+pub(super) mod context;
+pub(super) mod disassemble;
+pub(super) mod dot;
+pub(super) mod examine;
+pub(super) mod exit;
+pub(super) mod help;
+pub(super) mod label;
+pub(super) mod labels;
+pub(super) mod load;
+pub(super) mod print;
+pub(super) mod reset;
+pub(super) mod run;
+pub(super) mod step;
+pub(super) mod watchpoint;
 
 use super::{error::CommandResult, State};
 
+// TODO: remove once if-let chaining is in
+#[derive(Clone)]
+pub(crate) enum ArgumentKind {
+    Number(i64),
+    File(String),
+    Label(String),
+    Item(String),
+    Command(String),
+    SubCommand(String),
+    Any(String),
+}
```

Comment:
a trait would be better here. since then you also get completion and hinting for free cos you could call like `Trait::complete` for each command type